powerspinner/src/main/java/com/skydoves/powerspinner/PowerSpinnerView.kt

```diff
@@ -235,6 +235,7 @@ class PowerSpinnerView : AppCompatTextView, LifecycleObserver {
    */
   var lifecycleOwner: LifecycleOwner? = null
     set(value) {
+      field?.lifecycle?.removeObserver(this@PowerSpinnerView)
```

Comment:
We don't need to remove the previous observer here, because the internal hash map in the `LifecycleRegistry` checks whether the observer is already registered.

Reply:
It's not removing it from the new `lifecycleOwner` here. It's removing the observer from the previous lifecycleOwner (if it was not null).

Reply:
Yes, that's why we don't need to call `removeObserver` here. As you can see in the [LifecycleRegisitry](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:lifecycle/lifecycle-runtime/src/main/java/androidx/lifecycle/LifecycleRegistry.java;l=181?q=lifecyclere), it will not add into the lifecycle hash, which is handled in the `ComponentActivity` if there's any previous lifecycle is exist already.

Reply:
I'm not sure if we're talking about the same thing :)

on line 238 it calls `removeObserver` from `lifecycleOwnerA.lifecycle` (if `field` is not `null`). On line 240 it calls `addObserver` on `lifecycleOwnerB.lifecycle`.

Or am I just not getting what the LifecycleRegistery does?

Reply:
So the reason why you're calling the `removeObserver` is to remove the previous observer, which is registered in the internal cache before adding a duplicated observer right? The `addObserver` checks if there is the same observer is already registered in the internal hash. As a result, it can't be added duplicated observer instances into the internal hash, which means we don't need to call `removeObserver`.

Reply:
No, on line 238 it's removing the observer from the current `lifecycleOwner`. On line 239 a new `lifecycleOwner` is set. And on line 240 the observer is added to the `lifecyclerOwner` that was set on line 239.

So on line 238 and 240 we're not talking about the same lifecycleOwner. It has been updated.

Reply:
I have to sleep now :D I'll check this again tomorrow.

Reply:
I got it! That makes sense. Thank you for describing the details :)
When you back online, let's merge this. Thanks!